From bf350b8ccfdd2ee3a56a6438bf0689c7b122c4ca Mon Sep 17 00:00:00 2001
From: Deepali Chourasia <deepch23@in.ibm.com>
Date: Tue, 16 Aug 2022 10:42:46 +0000
Subject: [PATCH] disable

---
 build_tools/setup_helpers/extension.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/build_tools/setup_helpers/extension.py b/build_tools/setup_helpers/extension.py
index 1f7236e4..db47fc31 100644
--- a/build_tools/setup_helpers/extension.py
+++ b/build_tools/setup_helpers/extension.py
@@ -72,6 +72,7 @@ class CMakeBuild(build_ext):
             "-DBUILD_SHARED_LIBS=OFF",
             "-DCMAKE_POLICY_DEFAULT_CMP0063=NEW",
             "-DSPM_ENABLE_SHARED=OFF",
+            "-DSPM_ENABLE_TCMALLOC=OFF",
         ]
         build_args = ["--target", "install"]
 
-- 
2.34.1

